h2/stream.py

```diff
@@ -1040,18 +1040,26 @@ def _build_hdr_validation_flags(self, events):
             is_trailer = isinstance(
                 events[0], (_TrailersSent, TrailersReceived)
             )
+            is_response_header = isinstance(
+                events[0], (_ResponseHeadersSent, ResponseReceived)
```

Comment:
Rather than deal with this in this ticket, let's open a new one related to this issue and #314 that refactors the event names.